GitUI/UserControls/BlameControl.cs

```diff
@@ -641,8 +646,62 @@ private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventAr
             // Try get actual parent revision, get popup if it does not exist.
             // (The menu should be disabled if previous is not in grid).
             GitRevision? revision = _revisionGridInfo?.GetActualRevision(blameInfo.selectedRevision);
-            _clickedBlameLine = _lastBlameLine;
-            BlameRevision(revision?.FirstParentId, blameInfo.filename);
+            ObjectId parentId = revision?.FirstParentId;
+
+            int currentLineSelected = _lastBlameLine.OriginLineNumber;
+            int calculatedNewLineSelected = currentLineSelected;
+
+            if (currentLineSelected != 1)
```

Comment:
Why not thishandling when blaming "current" revision?

Reply:
OK that is working mostly, there are some situations though that fail but it may be due to selections. See it I investigate.

Reply:
Before investigating too much, check if it's not due to 'Detect moved or copied lined' which is enabled.